### ISSUE TITLE
Add support for GlobFS

### ISSUE THREAD

**Issue by jcchavezs** [yalue/merged_fs#3]
Currently the merged FS does not support Glob, this is inconvenient as some underlying FS might support it.

It would be cool if we could only return the Glob interface if both A and B implement it but that would require to either create another method or return the interface FS instead a concrete type (breaking change).

**Comment by yalue** [comment]
Thanks for the feedback!  True, it would be nice to have.  The hard part, of course, is supporting merging the results of Glob into a single list for patterns that match content in both underlying FS's.  Perhaps code like the following wouldn't be that bad:

```go
func (m *MergedFS) Glob(pattern string) ([]string, error) {
    contentsA, _ := fs.Glob(m.A, pattern)
    contentsB, _ := fs.Glob(m.B, pattern)
    
    // The key would be making this as efficient as possible
    results := mergePathLists(contentsA, contentsB)
    return results, nil
}
```

Using `fs.Glob` would take care of the optimized calls if either underlying FS implemented GlobFS.  The main issue is whether the "mergePathLists" code would be faster than just letting the default `fs.Glob` invoke the existing directory merging code.  Based on intuition, I'd expect it to be an improvement even with a naive implementation of `mergePathLists`, but I'd like to have a `BenchmarkGlobFSPerformance` test (or something like that) when adding such a feature.

I can probably add something like this in a few days, but I'd be happy to take any suggestions or PRs if you want to work on it yourself.

**Comment by yalue** [comment]
I was thinking about this more, and it's a bit more complicated than I realized when writing my earlier comment.  That snippet actually wouldn't work.  Glob could actually list contents of directories that ought not to be "visible."

Imagine the following scenario:
 - We are merging FSs A and B, with A having higher priority.
 - A contains a regular file, 'x'
 - B contains a directory x, with some content.  Say, 'x/1.txt', and 'x/2.txt'.
 - We call fs.Glob(merged, "x/*.txt")

The correct results of the pattern "x/*.txt" in the above scenario ought to be nothing at all, since the merged FS shows "x" as a regular file rather than a directory.  However, the implementation in my earlier comment would return "x/1.txt" and "x/2.txt"--files you wouldn't be able to Open in the merged FS.

I actually added a test case to check for this in commit c2f37d02e1b5bd0.  (`test_a.zip` contains a regular file named 'a', which is a directory in `test_b.zip`, and the new `TestGlob` contains code to ensure that `a/*` won't return the contents of the directory.

An optimization is still likely possible, but it would likely require checking each file to make sure it isn't in an unreachable directory, perhaps using the existing `validatePathPrefix` function.
